### PR TITLE
EPMDJ-5316: support topic "/build/summary" for dashboard page

### PR DESCRIPTION
**Pull request by SMPlay**
Issue: EPMDJ-5316

### CODE REVIEW

File: src/types/build-summary.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { ArrowType } from './arrow-type';
+import { Count } from './count';
+import { RisksSummary } from './risks-summary';
+import { TestCount } from './test-count';
+
+export interface BuildSummary {
```
Comment (IliaSolovev) [inline src/types/build-summary.ts:6]: are you sure that all these properties BE will give? Mb you`ll use optional chaining here

**Review by IliaSolovev — CHANGES_REQUESTED** [review]

File: src/types/build-summary.ts (hunk 1 above)
Comment (SMPlay) [reply src/types/build-summary.ts:6]: yes

File: src/pages/agent-full-page/dashboard/sections/risks-section.tsx [hunk 2]
```diff
@@ -1,46 +1,47 @@
...
+      info={total}
       graph={(
         <Tooltip message={<SectionTooltip data={tooltipData} hideValue />}>
           <Panel>
             <SingleBar
               width={64}
               height={128}
               color={RISKS_TYPES_COLOR.NEW}
-              percent={(newMethods.length / risksCount) * 100}
+              percent={(newMethodsCount / total) * 100}
```
Comment (RomanAchilov) [inline src/pages/agent-full-page/dashboard/sections/risks-section.tsx:37]: can be NaN

File: src/pages/agent-full-page/dashboard/sections/risks-section.tsx [hunk 3]
```diff
@@ -1,46 +1,47 @@
...
               icon={<Icons.Add height={16} width={16} />}
             />
             <SingleBar
               width={64}
               height={128}
               color={RISKS_TYPES_COLOR.MODIFIED}
-              percent={(modifiedMethods.length / risksCount) * 100}
+              percent={(modifiedMethodsCount / total) * 100}
```
Comment (RomanAchilov) [inline src/pages/agent-full-page/dashboard/sections/risks-section.tsx:44]: can be NaN

File: src/pages/agent-full-page/dashboard/sections/tests-to-run-section.tsx [hunk 4]
```diff
@@ -41,7 +36,7 @@ export const TestsToRunSection = () => {
                 width={64}
                 height={128}
                 color={TESTS_TO_RUN_TYPES_COLOR[testType as TestTypes]}
-                percent={(testsToRunInfo[testType as TestTypes].count / totalTestsToRunCount) * 100}
+                percent={(testsToRunByType[testType as TestTypes] / count) * 100}
```
Comment (RomanAchilov) [inline src/pages/agent-full-page/dashboard/sections/tests-to-run-section.tsx:39]: can be NaN

File: src/types/build-summary.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  coverage: number;
+  coverageCount: Count;
+  scopeCount: number;
+  arrow: ArrowType;
+  riskCounts: RisksSummary;
+  tests: TestCount;
+  testsToRun: TestCount;
+  recommendations: string[];
+}
```
Comment (RomanAchilov) [inline src/types/build-summary.ts:15]:
```suggestion
}
export interface BuildSummary {
  coverage?: number;
  coverageCount?: Count;
  scopeCount?: number;
  arrow?: ArrowType;
  riskCounts?: RisksSummary;
  tests?: TestCount;
  testsToRun?: TestCount;
  recommendations?: string[];
}
```

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/pages/agent-full-page/dashboard/sections/risks-section.tsx (hunk 2 above)
Comment (pavyarov) [reply src/pages/agent-full-page/dashboard/sections/risks-section.tsx:37]: Use convert util from #457

**Review by RomanAchilov — APPROVED** [review]

**Pull request merged**